SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -28,169 +30,191 @@ void USpatialTypebindingManager::Init(USpatialNetDriver* InNetDriver)
 			}
+		}
+	}
 
-			if (Property->PropertyFlags & CPF_AlwaysInterested)
-			{
-				for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
-				{
-					FInterestPropertyInfo InterestInfo;
-					InterestInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
-					InterestInfo.Property = Property;
+	ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
+	{
+		Worker_ComponentId ComponentId = SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SchemaComponents[Type];
+		if (ComponentId != 0)
+		{
+			Info.SchemaComponents[Type] = ComponentId;
+			ComponentToClassInfoMap.Emplace(ComponentId, MakeShared<FClassInfo>(Info));
+			ComponentToOffsetMap.Add(ComponentId, 0);
+			ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
+		}
+	});
 
-					Info.InterestProperties.Add(InterestInfo);
-				}
-			}
+	for (auto& SubobjectClassDataPair : SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SubobjectData)
+	{
+		int32 Offset = SubobjectClassDataPair.Key;
+		FSubobjectSchemaData SubobjectSchemaData = SubobjectClassDataPair.Value;
+
+		UClass* SubobjectClass = ResolveClass(SubobjectSchemaData.ClassPath);
+
+		FClassInfo* SubobjectInfoPtr = FindClassInfoByClass(SubobjectClass);
+		if (SubobjectInfoPtr == nullptr)
+		{
+			continue;
 		}
 
+		// Make a copy of the already made FClassInfo for this specific subobject
+		TSharedPtr<FClassInfo> ActorSubobjectInfo = MakeShared<FClassInfo>(*SubobjectInfoPtr);
+		ActorSubobjectInfo->SubobjectName = SubobjectSchemaData.Name;
+
 		ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
 		{
-			Worker_ComponentId ComponentId = SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SchemaComponents[Type];
+			Worker_ComponentId ComponentId = SubobjectSchemaData.SchemaComponents[Type];
 			if (ComponentId != 0)
 			{
-				Info.SchemaComponents[Type] = ComponentId;
-				ComponentToClassMap.Add(ComponentId, Class);
-				ComponentToOffsetMap.Add(ComponentId, 0);
-				ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
+				ActorSubobjectInfo->SchemaComponents[Type] = ComponentId;
+				ComponentToClassInfoMap.Emplace(ComponentId, ActorSubobjectInfo);
+				ComponentToOffsetMap.Add(ComponentId, Offset);
+				ComponentToCategoryMap.Add(ComponentId, ESchemaComponentType(Type));
 			}
 		});
 
-		Info.Class = Class;
+		// This is needed because the recursive call to FindClassInfoByClass for subobjects changes the map
+		Info = ClassInfoMap.FindChecked(Class);
+		Info.SubobjectInfo.Add(Offset, ActorSubobjectInfo);
+	}
+}
 
-		ClassInfoMap.Emplace(Class, Info);
+UClass* USpatialTypebindingManager::LoadClassForComponent(Worker_ComponentId ComponentId)
+{
+	// Only try to load classes for generated components
+	if (ComponentId < SpatialConstants::STARTING_GENERATED_COMPONENT_ID)
+	{
+		return nullptr;
 	}
 
-	for (UClass* Class : SupportedClasses)
+	for (auto& ClassDataPair : SchemaDatabase->ClassPathToSchema)
```

Comment:
Maybe I'm just noticing this now, but the runtime of this function is fairly bad. I think it might be worth creating a map ahead of time from component Id -> class path and using that for look ups.

Reply:
This could be done in schema generation and will probably help if performance is an issue.

Reply:
The map is going to be as large as any other maps containing component ids would be. I think it's potentially worth doing now since I don't like that we are iterating so much every time we load a new class.